metricflow/test/generate_snapshots.py

```diff
@@ -0,0 +1,110 @@
+"""Script to help generate test snapshots for all supported engines.
+
+Credentials are stored as a JSON string in an environment variable set via a shell command like:
+
+export MF_TEST_ENGINE_CREDENTIALS=$(cat <<EOF
+{
+    "duck_db": {
+        "engine_url": null,
+        "engine_password": null
+    },
+    "postgres": {
+        "engine_url": "postgresql://...",
+        "engine_password": "..."
+    },
+    "redshift": {
+        "engine_url": "redshift://...",
+        "engine_password": "..."
+    },
+    "snowflake": {
+        "engine_url": "snowflake://...",
+        "engine_password": "..."
+    },
+    "big_query": {
+        "engine_url": "bigquery://",
+        "engine_password": "..."
+    },
+    "databricks": {
+        "engine_url": "databricks://...",
+        "engine_password": "..."
+    }
+}
+EOF
+)
+"""
+from __future__ import annotations
+
+import logging
+import os
+from typing import Optional, Sequence
+
+from dbt_semantic_interfaces.implementations.base import FrozenBaseModel
+
+from metricflow.configuration.env_var import EnvironmentVariable
+
+logger = logging.getLogger(__name__)
+
+
+class MetricFlowTestCredentialSet(FrozenBaseModel):  # noqa: D
+    engine_url: Optional[str]
+    engine_password: Optional[str]
+
+
+class MetricFlowTestCredentialSetForAllEngines(FrozenBaseModel):  # noqa: D
+    duck_db: MetricFlowTestCredentialSet
+    postgres: MetricFlowTestCredentialSet
+    redshift: MetricFlowTestCredentialSet
+    snowflake: MetricFlowTestCredentialSet
+    big_query: MetricFlowTestCredentialSet
+    databricks: MetricFlowTestCredentialSet
+
+    @property
+    def as_sequence(self) -> Sequence[MetricFlowTestCredentialSet]:  # noqa: D
+        return (self.duck_db, self.postgres, self.redshift, self.snowflake, self.big_query, self.databricks)
+
+
+SNAPSHOT_GENERATING_TEST_FILES = (
+    "metricflow/test/dataflow/builder/test_dataflow_plan_builder.py",
+    "metricflow/test/plan_conversion/test_dataflow_to_sql_plan.py",
+    "metricflow/test/integration/test_rendered_query.py",
+)
```

Comment:
There's also tests here `metricflow/test/dataflow/optimizer/source_scan/` that have snapshot generating tests

Reply:
Added.

Reply:
https://github.com/dbt-labs/metricflow/blob/main/metricflow/test/model/test_data_warehouse_tasks.py#L229-L253

There are others missing too. Even my old -k filter (`pytest -k "dataflow or plan or optimizer or source_scan or test_rendered_query or test_build_metric_tasks" ...`) doesn't cover it all anymore.

Reply:
<img width="798" alt="Screenshot 2023-06-20 at 2 26 19 PM" src="https://github.com/dbt-labs/metricflow/assets/33618746/67fe4f51-9060-4019-93bb-8fe99be5d6da">
<img width="805" alt="Screenshot 2023-06-20 at 2 27 15 PM" src="https://github.com/dbt-labs/metricflow/assets/33618746/b67fa68d-28a8-4d48-b315-8d8ebeb0a290">
seems to be more actually with 
```
- test_rendered_query.py
- test_data_warehouse_tasks.py
- compare_sql_plan.py
```